src/pytest_workflow/plugin.py

```diff
@@ -66,45 +80,77 @@ def __init__(self, workflow_test: WorkflowTest, parent: pytest.Collector):
         self.workflow_test = workflow_test
         super().__init__(workflow_test.name, parent=parent)
 
-    def collect(self):
-        """This runs the workflow and starts all the associated tests
-        The idea is that isolated parts of the yaml get their own collector or
-        item."""
-
-        # Create a temporary directory where the workflow is run.
-        # This will prevent the project repository from getting filled up with
-        # test workflow output.
-        tempdir = tempfile.mkdtemp(prefix="pytest_wf")
+    def run_workflow(self):
+        """Runs the workflow in a temporary directory
+
+        Running in a temporary directory will prevent the project repository
+        from getting filled up with test workflow output.
+        The temporary directory is produced from self.config._tmpdirhandler
+        which  does the same as using a `tmpdir` fixture.
+
+        The temporary directory name is constructed from the test name by
+        replacing all whitespaces with '_'. Directory paths with whitespace in
+        them are very annoying to inspect.
+        Additionally the temporary directories are numbered. This prevents
+        name collisions if tests have the same name (when whitespace is
+        replaced). The alternative is overengineering some name collision
```

Comment:
Could you elaborate on the necessity of manually doing collision prevention?  Python's `tempfile` module contains functions and classes that guarantee there are no race conditions. Does pytest's `TempdirFactory` not use `tempfile`?

Reply:
No it does not. I have explained this more clearly now.